pkg/extensions/lint/lint.go

```diff
@@ -53,19 +53,53 @@ func (linter *Linter) CheckMandatoryAnnotations(repo string, manifestDigest godi
 		return false, err
 	}
 
-	annotations := manifest.Annotations
+	mandatoryAnnotationsMap := make(map[string]bool)
+	for _, annotation := range mandatoryAnnotationsList {
+		mandatoryAnnotationsMap[annotation] = false
+	}
+
+	manifestAnnotations := manifest.Annotations
+	for annotation := range manifestAnnotations {
+		if _, ok := mandatoryAnnotationsMap[annotation]; ok {
+			mandatoryAnnotationsMap[annotation] = true
+		}
+	}
+
+	missingAnnotations := getMissingAnnotations(mandatoryAnnotationsMap)
+	if len(missingAnnotations) == 0 {
+		return true, nil
+	}
 
-	for _, annot := range mandatoryAnnotationsList {
-		_, found := annotations[annot]
+	// if there are mandatory annotations missing in the manifest, get config and check these annotations too
+	configDigest := manifest.Config.Digest
+	content, err = imgStore.GetBlobContent(repo, string(configDigest))
 
-		if !found {
-			// if annotations are not found, return false but it's not an error
-			linter.log.Error().Msgf("linter: missing %s annotations", annot)
+	if err != nil {
+		linter.log.Error().Err(err).Msg("linter: couldn't get config JSON" + string(configDigest))
+
+		return false, err
+	}
+
+	if err := json.Unmarshal(content, &manifest); err != nil {
+		linter.log.Error().Err(err).Msg("linter: couldn't unmarshal config JSON")
 
-			return false, nil
+		return false, err
+	}
+
+	configAnnotations := manifest.Config.Annotations
```

Comment:
here you are looking on config descriptor annotations, which is found in the manifest, we need to look on config labels from config struct (not its descriptor)

look at my pr: https://github.com/project-zot/zot/pull/787

so you need to unmarshal the config blob in a ispec.ImageConfig var and look on its labels.

Right @andaaron ?